sympy/integrals/transforms.py

```diff
@@ -1008,9 +1008,13 @@ def repl(ex, *args):
 def _laplace_transform(f, t, s_, simplify=True):
     """ The backend function for Laplace transforms. """
     from sympy import (re, Max, exp, pi, Min, periodic_argument as arg_,
-                       arg, cos, Wild, symbols, polar_lift)
+                       arg, cos, Wild, symbols, polar_lift, limit, DiracDelta)
     s = Dummy('s')
-    F = integrate(exp(-s*t) * f, (t, 0, oo))
+    zero_minus = symbols('zero_minus',real=True)
+    if f.has(DiracDelta):
+        F = limit(integrate(exp(-s*t) * f, (t, zero_minus, oo)), zero_minus, 0, dir='-')
+    else:
+        F = integrate(exp(-s*t) * f, (t, 0, oo))
```

Comment:
Why does this do if f.has(DiracDelta)?

If we are saying that this is basically the definition of the Laplace transform then surely it would also apply when there is no Dirac delta function.

Reply:
> Why does this do if f.has(DiracDelta)?
> 
> If we are saying that this is basically the definition of the Laplace transform then surely it would also apply when there is no Dirac delta function.

The one-sided `laplace_transform` must work for all functions that behave well for `t>=0`, no matter how crazy they behave for negative `t`, as long as they are OK at infinitesimally small negative values.

I cannot be sure that `integrate(exp(-s*t) * f, (t, zero_minus, oo))` will always converge when `integrate(exp(-s*t) * f, (t, 0, oo))` converges, so I'd rather add an exception to be on the safe side. It also needs less computation time.

Unless there is a `DiracDelta` at `t==0`, it makes no difference anyway. If we ever encounter another delta-like distribution that Laplace needs to know about, we can add it too.

Reply:
If we're just making a special case for a Dirac delta at t=0 then it would be better just to compute the special case result rather than using `limit` since as you say there is no guarantee that the limit exists anyway.

Reply:
Great idea! That is very simple, and I tried a few things, it works well.